sys/kern/kern_mbuf.c

```diff
@@ -1543,6 +1543,7 @@ m_extadd(struct mbuf *mb, char *buf, u_int size, m_ext_free_t freef,
 	KASSERT(type != EXT_CLUSTER, ("%s: EXT_CLUSTER not allowed", __func__));
 
 	mb->m_flags |= (M_EXT | flags);
+	/* XXX-AM: Should we have an option to warn when these are not exact? */
```

Comment:
Is it common that size != cheri_getlen(buf) in these cases?

Reply:
You can have arbitrary buffers here in theory.  I just used this function yesterday to attach a 4K buffer to an mbuf in the NVMe Fabrics target code.  I also use it in the fabrics target code for I/O buffers from a LUN backend sent back over the socket.  In theory those can be sized as an arbitrary number of blocks (e.g. N * 512).  If N is a power of 2 those would all be fine, but possibly it could be a non-power-of-2 in which case you could end up with an inexact size.